src/libs/PolicyUtils.ts

```diff
@@ -575,6 +575,40 @@ function getIntegrationLastSuccessfulDate(connection?: Connections[keyof Connect
     return (connection as ConnectionWithLastSyncData)?.lastSync?.successfulDate;
 }
 
+function getSageIntacctVendors(policy: Policy | undefined, selectedVendorId: string | null | undefined): SelectorType[] {
+    const vendors = policy?.connections?.intacct?.data?.vendors ?? [];
+    const isMatchFound = vendors?.some(({id}) => id === selectedVendorId);
+
+    return (vendors ?? []).map(({id, value}) => ({
```

Comment:
You've already null-coalesced this.
```suggestion
    return (vendors).map(({id, value}) => ({
```